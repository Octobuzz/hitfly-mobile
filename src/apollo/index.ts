import * as _storage from './storage'
export const storage = _storage

export * from './queries'
export * from './schemas'
export * from './mutations'
export * from './commonTypes'

export { default as setupClient } from './setupClient'
