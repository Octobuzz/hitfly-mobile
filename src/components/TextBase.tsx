import styled from 'src/styled-components'
import React from 'react'

const TextBase = styled.Text`
  font-family: ${({ theme }) => theme.fonts.regular};
  font-size: 16px;
  line-height: 20px;
  color: ${({ theme }) => theme.colors.textMain};
`

export const HelperText = styled(props => <TextBase {...props} />)`
  color: ${({ theme }) => theme.colors.textAlt};
  text-align: center;
  margin-bottom: 40px;
`

export const H1 = styled(props => <TextBase {...props} />)`
  font-family: ${({ theme }) => theme.fonts.bold};
  font-size: 20px;
`

export const H2 = styled(props => <TextBase {...props} />)`
  font-family: ${({ theme }) => theme.fonts.bold};
  font-size: 16px;
`

export default TextBase
