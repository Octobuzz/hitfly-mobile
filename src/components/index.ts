export * from './views'
export * from './fields'
export * from './buttons'

export { default as TextBase } from './TextBase'
export { default as GenreItem } from './GenreItem'
export { default as TextWithLines } from './TextWithLines'
