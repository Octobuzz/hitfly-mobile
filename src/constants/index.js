import * as colors from './colors'
import * as images from './images'
import * as sizes from './sizes'
import style from './style'

export const appName = 'digicomob'

export { images, colors, sizes, style }
