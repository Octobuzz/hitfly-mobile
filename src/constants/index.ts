import * as storageKeys from './storageKeys'
import * as strings from './strings'
import * as colors from './colors'
import * as images from './images'
import * as styles from './styles'
import * as names from './names'

export { strings, images, colors, names, styles, storageKeys }
