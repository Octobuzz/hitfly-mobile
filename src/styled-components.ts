import * as styledComponents from 'styled-components/native'

export interface ThemeColors {
  brandBlue: string
  brandPink: string
  white: string
  textMain: string
  textAlt: string
  textGray: string
  inputBorder: string
}

export interface ThemeFonts {
  black: string
  bold: string
  medium: string
  regular: string
}

export interface ITheme {
  colors: ThemeColors
  fonts: ThemeFonts
}

const {
  default: styled,
  css,
  ThemeProvider,
} = styledComponents as styledComponents.ReactNativeThemedStyledComponentsModule<
  ITheme
>

export { css, ThemeProvider }
export default styled
