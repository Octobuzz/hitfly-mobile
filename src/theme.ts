import { ITheme } from './styled-components'

const theme: ITheme = {
  colors: {
    brandBlue: '#4297FE',
    brandPink: '#D066BF',
  },
}

export default theme
