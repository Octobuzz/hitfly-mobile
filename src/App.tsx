import React from 'react'
import './reactotron.config'
import { YellowBox } from 'react-native'
import { ThemeProvider } from 'styled-components'
import ApolloClient, { InMemoryCache } from 'apollo-boost'
import { persistCache } from 'apollo-cache-persist'
import { ApolloProvider } from '@apollo/react-hooks'
import AppNavigator from './navigation'
import { storageInstance, storage } from './utils'
import theme from './theme'
import { names, storageKeys } from './constants'

// либы до сих пор используют это
YellowBox.ignoreWarnings([
  'Warning: componentWillMount is deprecated',
  'Warning: componentWillUpdate is deprecated',
  'Warning: componentWillReceiveProps is deprecated',
])

async function createApolloClient(): Promise<ApolloClient<InMemoryCache>> {
  const cache = new InMemoryCache({})

  await persistCache({
    cache,
    // @ts-ignore
    storage: storageInstance,
  })

  const token = await storage.getItem(storageKeys.AUTH_TOKEN)

  const client = new ApolloClient<InMemoryCache>({
    cache,
    headers: {
      'X-TOKEN-AUTH': token,
    },
    uri: names.BASE_URL,
  })

  return client
}

class App extends React.Component<{}, { isReady: boolean }> {
  client?: ApolloClient<InMemoryCache>
  state = {
    isReady: false,
  }

  async componentDidMount() {
    this.client = await createApolloClient()
    this.setState({ isReady: true })
  }

  render() {
    const { isReady } = this.state
    return (
      isReady &&
      this.client && (
        <ApolloProvider client={this.client}>
          <ThemeProvider theme={theme}>
            <AppNavigator />
          </ThemeProvider>
        </ApolloProvider>
      )
    )
  }
}

export default App
