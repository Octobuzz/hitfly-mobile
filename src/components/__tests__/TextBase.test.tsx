import React from 'react'
import TextBase from '../TextBase'
import { ThemeProvider } from 'src/styled-components'
import renderer from 'react-test-renderer'
import theme from 'src/theme'

test('renders correctly', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <TextBase>test text</TextBase>
      </ThemeProvider>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
