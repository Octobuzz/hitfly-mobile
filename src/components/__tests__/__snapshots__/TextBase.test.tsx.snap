// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#313131",
        "fontFamily": "GothamPro",
        "fontSize": 16,
        "lineHeight": 20,
      },
    ]
  }
>
  test text
</Text>
`;
